packages/ns-api/files/ns.devices

```diff
@@ -0,0 +1,997 @@
+               not (is_bridge(physical_dev) and not get_interface(
+                   physical_dev.get('name'), ifaces_from_config)) and
+               # discard unconfigured bond devices (they would appear after committing bond creation)
+               not is_bond(physical_dev),
+               physical_devices))
+
+    # add data from config
+
+    merged_devices = []
+
+    for physical_dev in filtered_physical_devices:
+        device_from_config_found = next((device_from_config for device_from_config in devices_from_config if physical_dev.get(
+            'name') == device_from_config.get('name')), None)
+
+        if device_from_config_found:
+            merged_devices.append(device_from_config_found | physical_dev)
+        else:
+            merged_devices.append(physical_dev)
+
+    # add remaining devices from config
+
+    for device_from_config in devices_from_config:
+        dev_found = next((dev for dev in merged_devices if dev.get(
+            'name') == device_from_config.get('name')), None)
+
+        if not dev_found:
+            merged_devices.append(device_from_config)
+
+    all_devices = merged_devices + ifaces_without_device
+
+    # do not show devices used by bridges or bonds
+
+    devices_used_by_logical_ifaces = []
+
+    for device in all_devices:
+        if is_bond(device):
+            devices_used_by_logical_ifaces += device.get('slaves')
+        elif is_bridge(device):
+            devices_used_by_logical_ifaces += device.get('ports')
+
+    devices_to_show = list(filter(lambda device: not get_name(
+        device) in devices_used_by_logical_ifaces, all_devices))
+
+    # retrieve stats
+
+    stats = get_device_stats()
+
+    for stat in stats:
+        dev_found = next((device for device in devices_to_show if device.get(
+            'name') == stat.get('ifname')), None)
+
+        if dev_found:
+            dev_found['stats'] = {
+                'collisions': stat.get('stats64').get('tx').get('collisions'),
+                'multicast': stat.get('stats64').get('rx').get('multicast'),
+                'rx_bytes': stat.get('stats64').get('rx').get('bytes'),
+                'rx_dropped': stat.get('stats64').get('rx').get('dropped'),
+                'rx_errors': stat.get('stats64').get('rx').get('errors'),
+                'rx_packets': stat.get('stats64').get('rx').get('packets'),
+                'tx_bytes': stat.get('stats64').get('tx').get('bytes'),
+                'tx_dropped': stat.get('stats64').get('tx').get('dropped'),
+                'tx_errors': stat.get('stats64').get('tx').get('errors'),
+                'tx_packets': stat.get('stats64').get('tx').get('packets'),
+            }
+
+    # retrieve devices speed
+
+    for device in devices_to_show:
+        try:
+            res = subprocess.run(
+                ['/bin/cat', f'/sys/class/net/{device.get("name")}/speed'], check=True, text=True, capture_output=True)
+            speed = int(res.stdout.strip())
```

Comment:
This is slow due to the context switch: it's better to create a a function which reads the file using python without the subprocess call